### PR TITLE
[IncidentSender] Avoid to record incidentHistory when using Redmine incident server

### PR DESCRIPTION
**Pull request by cosmo0920**
Related to https://github.com/project-hatohol/hatohol/issues/2263.

Cc: @ashie

### CODE REVIEW

File: server/src/IncidentSender.cc [hunk 1]
```diff
@@ -111,6 +111,7 @@ struct IncidentSender::Impl
 	AtomicValue<bool> trackerChanged;
 	std::mutex   trackerLock;
 	HatoholError lastResult;
+	bool shouldRecordIncidentHistory;
```
Comment (kz0817) [inline server/src/IncidentSender.cc]: Let's set the default vlaue at the constructor's initializer list.

Comment (cosmo0920) [inline server/src/IncidentSender.cc]: I've added it: https://github.com/project-hatohol/hatohol/pull/2266/files#diff-ab1d8e32a3713e346c9c43936482ca31R120

**Comment by Mnakagawa** [comment]
OK.
If success the tests, I merge this PR.

**Comment by cosmo0920** [comment]
Thanks!

**Pull request merged**